packages/opentelemetry-instrumentation-agno/opentelemetry/instrumentation/agno/__init__.py

```diff
@@ -0,0 +1,425 @@
+                    duration,
+                    attributes={
+                        GenAIAttributes.GEN_AI_SYSTEM: "agno",
+                        SpanAttributes.TRACELOOP_SPAN_KIND: TraceloopSpanKindValues.AGENT.value,
+                    }
+                )
+
+                return result
+
+            except Exception as e:
+                span.set_status(Status(StatusCode.ERROR, str(e)))
+                span.record_exception(e)
+                raise
+
+
+class _AgentARunWrapper:
+    """Wrapper for Agent.arun() method to capture asynchronous agent execution."""
+
+    def __init__(self, tracer, duration_histogram, token_histogram):
+        """Initialize the wrapper with OpenTelemetry instrumentation objects."""
+        self._tracer = tracer
+        self._duration_histogram = duration_histogram
+        self._token_histogram = token_histogram
+
+    @dont_throw
+    async def __call__(self, wrapped, instance, args, kwargs):
+        """Wrap the Agent.arun() call with tracing instrumentation."""
+        if context_api.get_value(
+            context_api._SUPPRESS_INSTRUMENTATION_KEY
+        ) or context_api.get_value("suppress_agno_instrumentation"):
+            return await wrapped(*args, **kwargs)
+
+        span_name = f"agno.agent.{getattr(instance, 'name', 'unknown')}"
+
+        with self._tracer.start_as_current_span(
+            span_name,
+            kind=SpanKind.CLIENT,
+        ) as span:
+            try:
+                span.set_attribute(GenAIAttributes.GEN_AI_SYSTEM, "agno")
+                span.set_attribute(SpanAttributes.TRACELOOP_SPAN_KIND,
+                                   TraceloopSpanKindValues.AGENT.value)
+
+                if hasattr(instance, 'name'):
+                    span.set_attribute(GenAIAttributes.GEN_AI_AGENT_NAME, instance.name)
+
+                if hasattr(instance, 'model') and instance.model:
+                    model_name = getattr(instance.model, 'id',
+                                         getattr(instance.model, 'name', 'unknown'))
+                    span.set_attribute(GenAIAttributes.GEN_AI_REQUEST_MODEL,
+                                       model_name)
+
+                if args and should_send_prompts():
+                    input_message = str(args[0])
+                    span.set_attribute(
+                        f"{GenAIAttributes.GEN_AI_PROMPT}.0.content",
+                        input_message)
+
+                import time
+                start_time = time.time()
+
+                result = await wrapped(*args, **kwargs)
+
+                duration = time.time() - start_time
+
+                if hasattr(result, 'content') and should_send_prompts():
+                    span.set_attribute(
+                        f"{GenAIAttributes.GEN_AI_COMPLETION}.0.content",
+                        str(result.content))
+
+                if hasattr(result, 'run_id'):
+                    span.set_attribute("agno.run.id", result.run_id)
```

Comment:
do you think it should be with an `agno` prefix? or an `agent` prefix? to consolidate different framworks